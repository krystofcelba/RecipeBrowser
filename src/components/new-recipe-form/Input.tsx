import React from 'react';
import { StyleSheet, TextInput, StyleProp, TextStyle, Keyboard } from 'react-native';

import AppTheme from '../../assets/appTheme';

interface Props {
  placeholder: string;
  numberOfLines?: number;
  value: string;
  onChangeText: (text: string) => void;
  style?: StyleProp<TextStyle>;
}

const Input = ({ placeholder, numberOfLines = 1, value, onChangeText, style, ...otherProps }: Props) => (
  <TextInput
    value={value}
    onChangeText={onChangeText}
    numberOfLines={numberOfLines}
    style={[styles.input, { height: numberOfLines * AppTheme.defaultFontSize }, style && style]}
    placeholder={placeholder}
    multiline={numberOfLines > 1}
    onSubmitEditing={Keyboard.dismiss}
    {...otherProps}
  />
);

const styles = StyleSheet.create({
  input: {
    fontFamily: AppTheme.fontFamily,
    fontSize: AppTheme.defaultFontSize,
  },
});

export default Input;
