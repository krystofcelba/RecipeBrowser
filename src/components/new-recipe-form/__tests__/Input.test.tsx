import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

describe('Input component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Input placeholder="placeholder" value="" onChangeText={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with defined style', () => {
    const wrapper = shallow(
      <Input placeholder="placeholder" value="" onChangeText={() => {}} style={{ margin: 10 }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('call onChangeText as expected', () => {
    const mockedText = 'a';
    const wrapper = shallow(
      <Input
        placeholder="placeholder"
        value=""
        onChangeText={text => {
          expect(text).toBe(mockedText);
        }}
        style={{ margin: 10 }}
      />,
    );
    wrapper.simulate('changeText', mockedText);
  });
});
