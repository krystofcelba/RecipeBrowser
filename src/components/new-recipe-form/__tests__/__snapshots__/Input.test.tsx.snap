// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders as expected 1`] = `
<TextInput
  multiline={false}
  numberOfLines={1}
  onChangeText={[Function]}
  onSubmitEditing={[Function]}
  placeholder="placeholder"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "height": 14,
      },
      undefined,
    ]
  }
  value=""
/>
`;

exports[`Input component renders as expected with defined style 1`] = `
<TextInput
  multiline={false}
  numberOfLines={1}
  onChangeText={[Function]}
  onSubmitEditing={[Function]}
  placeholder="placeholder"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "height": 14,
      },
      Object {
        "margin": 10,
      },
    ]
  }
  value=""
/>
`;
